django/utils/datastructures.py

```diff
@@ -114,6 +119,31 @@ def __setstate__(self, obj_dict):
             self.setlist(k, v)
         self.__dict__.update(obj_dict)
 
+    def pop(self, key, default=NOT_PASSED):
+        """
+        Remove and return the last data value for the passed key. If key doesn't exist
+        or value is an empty list, return `default`.
+        """
+        return self._pop(key, last=True, default=default)
+
+    def poplist(self, key, default=NOT_PASSED):
+        """
+        Remove and return the list of values for the key. If key doesn't exist, return a
+        default value.
+        """
+        return self._pop(key, last=False, default=default)
+
+    def _pop(self, key, last, default=NOT_PASSED):
```

Comment:
I am just a beginner was looking into pull requests to understand open source so this might be a bad question. Why does the default parameter has NOT_PASSED as its default value when all the methods that are using the method _pop already have that as their default value? For redundancy?